x-pack/plugins/actions/server/lib/is_conector_deprecated.ts

```diff
@@ -18,11 +18,39 @@ export const isConnectorDeprecated = (
      * Connectors after the Elastic ServiceNow application use the
      * Import Set API (https://developer.servicenow.com/dev.do#!/reference/api/rome/rest/c_ImportSetAPI)
      * A ServiceNow connector is considered deprecated if it uses the Table API.
-     *
-     * All other connectors do not have the usesTableApi config property
-     * so the function will always return false for them.
      */
-    return !!connector.config?.usesTableApi;
+
+    /**
+     * We cannot deduct if the connector is
+     * deprecated without config. In this case
+     * we always return false.
+     */
+    if (connector.config == null) {
+      return false;
+    }
+
+    /**
+     * If the usesTableApi is not defined it means that the connector is created
+     * before the introduction of the usesTableApi property. In that case, the connector is assumed
+     * to be deprecated because all connectors prior 7.16 where using the Table API.
+     * Migrations x-pack/plugins/actions/server/saved_objects/actions_migrations.ts set
+     * the usesTableApi property to true to all connectors prior 7.16. Pre configured connectors
+     * cannot be migrated. This check ensures that pre configured connectors without the
+     * usesTableApi property explicitly in the kibana.yml file are considered deprecated.
+     * According to the schema defined here x-pack/plugins/actions/server/builtin_action_types/servicenow/schema.ts
+     * if the property is not defined it will be set to true at the execution of the connector.
+     */
+    if (connector.config.usesTableApi == null) {
```

Comment:
Same comment as above.
A `Object.hasOwnProperty` is probably the right approach here.